Dockerfile.windows

```diff
@@ -0,0 +1,70 @@
+# escape=`
+
+ARG WINDOWS_VERSION
+
+#
+# Builder Image - Windows Server Core
+# Assemble the components required to run Fluent Bit
+#
+FROM mcr.microsoft.com/windows/servercore:$WINDOWS_VERSION as builder
+
+# The FLUENTBIT_VERSION ARG must be after the FROM instruction
+ARG FLUENTBIT_VERSION
+ARG IMAGE_CREATE_DATE
+ARG IMAGE_SOURCE_REVISION
+
+# Metadata as defined in OCI image spec annotations - https://github.com/opencontainers/image-spec/blob/master/annotations.md
+LABEL org.opencontainers.image.title="Fluent Bit" `
+      org.opencontainers.image.description="Fluent Bit is an open source and multi-platform Log Processor and Forwarder which allows you to collect data/logs from different sources, unify and send them to multiple destinations. It's fully compatible with Docker and Kubernetes environments." `
+      org.opencontainers.image.created=$IMAGE_CREATE_DATE `
+      org.opencontainers.image.version=$FLUENTBIT_VERSION `
+      org.opencontainers.image.authors="Eduardo Silva <eduardo@treasure-data.com>" `
+      org.opencontainers.image.url="https://hub.docker.com/r/fluent/fluent-bit" `
+      org.opencontainers.image.documentation="https://docs.fluentbit.io/manual/" `
+      org.opencontainers.image.vendor="Fluent Organization" `
+      org.opencontainers.image.licenses="Apache-2.0" `
+      org.opencontainers.image.source="https://github.com/fluent/fluent-bit" `
+      org.opencontainers.image.revision=$IMAGE_SOURCE_REVISION
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+RUN Write-Host ('Creating folders'); `
+    New-Item -Type Directory -Path /installation; `
+    New-Item -Type Directory -Path /fluent-bit;
+
+COPY ./ci/appveyor-download.ps1 /installation/appveyor-download.ps1
+WORKDIR /installation
+
+RUN Write-Host ('Installing Fluent Bit'); `
+    /installation/appveyor-download.ps1 "v$($env:FLUENTBIT_VERSION)" -platform x64 -filetype zip; `
+    Expand-Archive -Path /installation/td-agent-bit-$($env:FLUENTBIT_VERSION)-win64.zip -Destination /installation/fluent-bit; `
+    Move-Item -Path /installation/fluent-bit/*/* -Destination /fluent-bit/;
+
+RUN Write-Host ('Installing required OpenSSL 1.0.2s libraries - ssleay32.dll, libeay32.dll.'); `

```

Comment:
I'm kinda worried by this third-party dependency, especially considering
that we really do not need this anymore since v1.4.0 (the dependency was
removed by #1832).

Yes, we still need OpenSSL to make v1.3.x series work properly, but what
do you think about removing this part?

Reply:
We need the OpenSSL dlls for v1.3.x, so I'm not sure what the alternative is at the moment? And it looks like you may need them again in v1.4.x if you bring back librdkafka.

Is there a way to build these and include them as part of the release package for fluentbit? So that we can remove this step here and move it to the build pipeline?

Reply:
> And it looks like you may need them again in v1.4.x if you bring back librdkafka.

That's correct. We need them if we want out_kafka even on v1.4.

> Is there a way to build these and include them as part of the release package for fluentbit?

No, I'm not aware there is. The ideal is to find some way to get librdkafka
work without OpenSSL, and make our binary work without any pre-requirement,
but we are not getting there yet.

OK, then we can leave this part of the script as is. We'll revisit this issue after
investigating more on librdkafka.

Reply:
@edsiper Or can you bring b0191b68 to v1.3 branch? This should allow
us to ship Dockerfile without thirdparty dependency.

This third-pardy binary is linked from OpenSSL wiki, but clearly labeled
as non-official. So there might be 

> Use these OpenSSL derived products at your own risk; these products have not been evaluated or tested by the OpenSSL project. 
> https://wiki.openssl.org/index.php/Binaries

I am wary of this. Especially it might have security implication for our
Windows users.

Reply:
@edsiper Or can you bring b0191b68 to v1.3 branch? It should enable
us to ship Dockerfile without third-party dependency.

This third-pardy binary is linked from OpenSSL wiki, but clearly labeled
as non-official. 

> Use these OpenSSL derived products at your own risk; these products have not been evaluated or tested by the OpenSSL project. 
> https://wiki.openssl.org/index.php/Binaries

I am wary of this. Especially this *might* have security implication for our
Windows users.

Reply:
@fujimotos I totally agree on the security implication concern. If we can pull in the removal of out_kafka into the 1.3 branch, then I can remove this dependency. Longer term there will need to be another solution once the out_kafka capability comes back in.

Reply:
> If we can pull in the removal of out_kafka into the 1.3 branch, then I can remove this dependency. Longer term there will need to be another solution once the out_kafka capability comes back in.

@paulbouwer OK, then let's go that route. So please update this patch:

1. Make Dockerfile fetch from fluentbit.io instead of AppVeyor.
2. Remove the code to pull the unofficial OpenSSL library.

Once these points are fixed, I have no objection to merge this PR into main.

Reply:
@paulbouwer We got the concerning patch ported to v1.3 branch in 616df472
(thanks to @edsiper) so we can now safely remove the dependency.